dev/NumberBox/NumberBox.cpp

```diff
@@ -642,3 +617,50 @@ bool NumberBox::IsInBounds(double value)
     return (value >= Minimum() && value <= Maximum());
 }
 
+void NumberBox::UpdateHeaderPresenterState(winrt::IInspectable const& header)
+{
+    // Load header presenter as late as possible
+    if(m_headerPresenter == nullptr)
+    {
+        if (const auto headerPresenter = GetTemplateChildT<winrt::ContentPresenter>(c_numberBoxHeaderName, (winrt::IControlProtected)*this))
+        {
+            // Set presenter to enable lightweight styling of the headers margin
+            m_headerPresenter.set(headerPresenter);
+        }
+    }
+
+    // To enable lightweight styling, collapse header presenter if there is no header specified
+    if (const auto headerPresenter = m_headerPresenter.get())
+    {
+        if (header != nullptr)
+        {
+            // Check if header is string or not
+            if (const auto headerAsString = Header().try_as<winrt::IReference<winrt::hstring>>())
+            {
+                if (headerAsString.Value().empty())
+                {
+                    // String is the empty string, hide presenter
```

Comment:
If `Header` is an empty string and `HeaderTemplate` is not null, the header presenter will be incorrectly made collapsed.

Reply:
Good catch. The number of conditions can probably be reduced if we do something like so.

auto headerVisibility = visible;
if((header is null && headerPresenter is null) ||
   (header is empty string) )
{
    headerVisibility = collapsed
} 

headerPresenter.visibility(headerVisibility)

Reply:
I've solved that now, I am not sure if your proposed change would be good to show what our idea is: Regardless of the header property, if there is a HeaderTemplate we will render it.